Tests/Functional/PhpcrOdmRepositoryTest.php

```diff
@@ -12,68 +12,47 @@
+    protected function getRepository()
     {
-        $this->dm = $this->db('PHPCR')->getOm();
-
-        $this->db('PHPCR')->purgeRepository(true);
-        $this->db('PHPCR')->createTestNode();
-
-        $rootDocument =  $this->dm->find(null, '/test');
-        $document = new Generic();
-        $document->setNodeName('foo');
-        $document->setParentDocument($rootDocument);
-        $this->dm->persist($document);
-
-        $document = new Generic();
-        $document->setNodeName('bar');
-        $document->setParentDocument($rootDocument);
-        $this->dm->persist($document);
-        $this->dm->flush();
-
-        $this->repositoryRegistry = $this->container->get('cmf_resource.registry');
-    }
-
-    public function provideGet()
-    {
-        return array(
-            array('/foo', 'foo'),
-            array('/bar', 'bar'),
-            array('/', 'test'),
-        );
+        return $this->repositoryRegistry->get('test_repository_phpcr_odm');
     }
 
     /**
-     * @dataProvider provideGet
+     * @dataProvider provideAdd
      */
-    public function testRepositoryGet($path, $expectedName)
+    public function testRepositoryAddSingleResource($path, $name)
     {
-        $repository = $this->repositoryRegistry->get('test_repository');
-        $res = $repository->get($path);
-        $this->assertNotNull($res);
-        $document = $res->getPayload();
-
-        $this->assertEquals($expectedName, $document->getNodeName());
-    }
+        $document = new Generic();
+        $document->setNodeName($name);
+        $document->setParentDocument($this->dm->find(null, '/test'.('/' === $path ? '' : $path)));
+        $this->dm->persist($document);
 
-    public function provideFind()
-    {
-        return array(
-            array('/*', 2),
-            array('/', 1),
-        );
+        $resource = new PhpcrOdmResource($path, $document);
+        $this->getRepository()->add($path, $resource);
+        
+        $document = $this->dm->find(null, '/test'.$path.('/' === $path ? '' : '/').$name);
+        $this->assertEquals($name, $document->getNodeName());
     }
 
     /**
-     * @dataProvider provideFind
+     * @dataProvider provideAdd
      */
-    public function testRepositoryFind($pattern, $nbResults)
+    public function testRepositoryAddResourceCollection($path, $name)
     {
-        $repository = $this->repositoryRegistry->get('test_repository');
-        $res = $repository->find($pattern);
-        $this->assertCount($nbResults, $res);
+        $document = new Generic();
+        $document->setNodeName($name);
+        $document->setParentDocument($this->dm->find(null, '/test'.('/' === $path ? '' : $path)));
+        $this->dm->persist($document);
+
+        $resource = new ArrayResourceCollection([new PhpcrOdmResource($path, $document)]);
+        $this->getRepository()->add($path, $resource);
+
+        $document = $this->dm->find(null, '/test'.$path.('/' === $path ? '' : '/').$name);
```

Comment:
actually symfony codestyle says to not have spaces in concatenation.

there is the PathHelper. you could do `'/test'.PathHelper::absolutizePath($name, $path)`